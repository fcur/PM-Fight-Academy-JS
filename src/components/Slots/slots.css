/* SLOTS */

.slotsContainer {
  display: flex;
  flex-direction: column;
}

.formContainer {
  order: 1;
  margin: auto;
}

.calendarContainer {
  order: 2;
}

/* CALENDAR */

.calendarContainer {
  display: flex;
  flex-direction: column;
  margin: auto;
}

.calendarBtns {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 820px;
}

.calendarBtns > * {
  outline: none;
  border: none;
  background: none;
  user-select: none;

  padding: 0.4em 4em;
  margin: 15px 0;
  height: fit-content;

  border-radius: 15px;
  box-sizing: border-box;
  background: #86868621;

  text-decoration: none;
  font-family: "Roboto", sans-serif;
  font-weight: 300;
  font-size: 1.3rem;
  color: #ffffff;
  text-align: center;
  transition: all 0.2s;
}

.calendarBtns > *:hover {
  color: #000000;
  background-color: #f8ff13;
  border-color: #f8ff13;

  cursor: pointer;
}

.calendar {
  display: flex;
  overflow-y: hidden;
  list-style: none;
  color: #fff;
  margin-bottom: 30px;
  width: 820px;
}

.calendar > * {
  width: 100px;
}

.hours {
  list-style: none;
  width: fit-content;
  align-self: flex-start;
  height: 100vh;
  margin: 40px 0;
  padding: 0;
  color: #fff;
}

.hours > * {
  margin-top: 5px;
  margin-right: 5px;
  height: 30px;
}

.calendar .dayColumn {
  padding: 0 5px;
  border-left: 1px solid rgba(255, 255, 255, 0.404);
}

.calendar .cell {
  margin-top: 5px;
  height: 30px;
  color: #000;
  background-color: #6d6d6d3a;
  border-radius: 10px;
}

.calendar .cell:hover {
  cursor: pointer;
  background-color: #74771863;
}

.clicked {
  background: red;
}

.calendar .day {
  background: none;
  color: #fff;
  border-radius: 0;
  border-bottom: 1px solid rgba(255, 255, 255, 0.404);
}

.calendar .day:hover {
  background: none;
  cursor: default;
}

.calendar .active {
  background-color: #f8ff13;
}

.calendar .active:hover {
  background: #f7ff13cc;
}

/* FORM */

.formContainer form {
  display: flex;
  flex-direction: row;
  gap: 10px;
  margin: auto;
}

.formContainer form button {
  width: fit-content;
  padding: 0.2em 2em 0.2em 1em;
}

select {
  outline: none;
  border: none;
  background: none;
  cursor: pointer;

  padding: 0.4em 4em;
  margin: 15px 0;
  height: fit-content;

  border-radius: 15px;
  border: 1px solid #f8ff13;
  box-sizing: border-box;

  font-size: 1.3rem;
  color: #f8ff13;
  text-align: center;
}

select * {
  color: #000;
}

option:hover {
  box-shadow: 0 0 10px 100px #f8ff13 inset;
  background: #f8ff13;
  color: #000;
}

@media screen and (max-width: 1225px) {
  .slotsContainer {
    display: flex;
    flex-direction: column;
  }

  .hours {
    margin-left: 0;
  }

  .calendarContainer,
  .calendarBtns {
    margin: auto;
    width: 80vw;
  }

  .calendar {
    max-width: fit-content;
    width: 80vw;
    margin: auto;
  }

  .calendarBtns * {
    width: fit-content;
    padding: 0.4em 2.7em;
  }

  .formContainer {
    margin: auto;
    width: 80vw;
  }
}

@media screen and (max-width: 760px) {
  .formContainer form {
    flex-direction: column;
    width: 85vw;
  }

  .formContainer form button {
    width: 50%;
    margin-top: 0;
  }

  .calendarContainer {
    width: 100vw;
  }

  .calendarBtns {
    margin: auto;
  }

  .calendarBtns * {
    padding: 0.4em 1.5em;
  }

  .calendar {
    margin: auto;
  }
}
