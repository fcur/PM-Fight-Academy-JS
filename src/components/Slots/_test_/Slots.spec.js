import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import Slots from "../Slots";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore();

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: () => [
    {
      id: 1,
      coachId: 1,
      dateStart: "test",
      timeStart: "test",
      timeEnd: "test",
    },
  ],
}));
describe("Slots", () => {
  it("should render properly", () => {
    const { container } = render(
      <Provider store={store}>
        <Slots />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
