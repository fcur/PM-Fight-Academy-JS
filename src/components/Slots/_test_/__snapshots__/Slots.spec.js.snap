// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slots should render properly 1`] = `
<div>
  <div
    class="slotsContainer"
  >
    <div
      class="calendarContainer"
    >
      <div
        class="calendarBtns"
      >
        <button>
          Back
        </button>
        <button>
          Today
        </button>
        <button>
          Next
        </button>
      </div>
      <div
        class="calendar"
      >
        <ul
          class="hours"
        >
          <li>
            0
            :00
          </li>
          <li>
            1
            :00
          </li>
          <li>
            2
            :00
          </li>
          <li>
            3
            :00
          </li>
          <li>
            4
            :00
          </li>
          <li>
            5
            :00
          </li>
          <li>
            6
            :00
          </li>
          <li>
            7
            :00
          </li>
          <li>
            8
            :00
          </li>
          <li>
            9
            :00
          </li>
          <li>
            10
            :00
          </li>
          <li>
            11
            :00
          </li>
          <li>
            12
            :00
          </li>
          <li>
            13
            :00
          </li>
          <li>
            14
            :00
          </li>
          <li>
            15
            :00
          </li>
          <li>
            16
            :00
          </li>
          <li>
            17
            :00
          </li>
          <li>
            18
            :00
          </li>
          <li>
            19
            :00
          </li>
          <li>
            20
            :00
          </li>
          <li>
            21
            :00
          </li>
          <li>
            22
            :00
          </li>
          <li>
            23
            :00
          </li>
        </ul>
        <div
          class="dayColumn"
        >
          <div
            class="day cell"
          >
            29.3
          </div>
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
        </div>
        <div
          class="dayColumn"
        >
          <div
            class="day cell"
          >
            30.3
          </div>
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
        </div>
        <div
          class="dayColumn"
        >
          <div
            class="day cell"
          >
            31.3
          </div>
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
        </div>
        <div
          class="dayColumn"
        >
          <div
            class="day cell"
          >
            1.4
          </div>
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
        </div>
        <div
          class="dayColumn"
        >
          <div
            class="day cell"
          >
            2.4
          </div>
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
        </div>
        <div
          class="dayColumn"
        >
          <div
            class="day cell"
          >
            3.4
          </div>
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
        </div>
        <div
          class="dayColumn"
        >
          <div
            class="day cell"
          >
            4.4
          </div>
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
          <div
            class="cell"
          />
        </div>
      </div>
    </div>
    <div
      class="formContainer"
    >
      <select>
        <option
          value="1"
        >
           
        </option>
      </select>
      <div
        class="btnContainer"
      >
        <button
          class="addBtn"
        >
          <p
            class="addBtnText"
          >
            Add
          </p>
        </button>
        <button
          class="addBtn"
        >
          <p
            class="addBtnText"
          >
            Delete
          </p>
        </button>
      </div>
    </div>
  </div>
</div>
`;
